### ISSUE TITLE
Ambient cross version compatibility

### ISSUE THREAD

**Issue by Stevenjin8** [istio/istio#52701]
From [Ambient GA RFC](https://docs.google.com/document/d/1GiD2-MmY9VrkfEs-ET0b7UGPmVII2GQUD7m1iAvbrjo):
> For proxy skew with Ztunnel and waypoints. This is likely already done, just need to make sure our documentation and testing is high enough quality.

**Comment by linsun** [comment]
@therealmitchconnors could be a good owner per @keithmattix , WDYT?

**Comment by therealmitchconnors** [comment]
Happy to own this.

**Comment by howardjohn** [comment]
We don't think we have skew between Istiod <-> {Ztunnel,waypoint} or CNI <-> Ztunnel which is probably top priority

**Comment by therealmitchconnors** [comment]
Working on this now.  Planned test flow looks like:

1. Upgrade CNI to version n+1
2. Create a pod, wait for ready
3. Delete a pod
4. Run connectivity test

Does that look correct for the CNI compatibility?  

I'm not sure under what circumstances a waypoint would need compatibility with a different Istiod, since the deployment is controlled by Istiod...  For Ztunnel/Istiod compat, do I just run connectivity tests, or do I need to look into WDS?

**Comment by howardjohn** [comment]
For CNI compat that looks right. Maybe also unenroll and re-enroll the pod as well would be 'extra credit', we have had bugs around that before.

We will also want ztunnel compat test that is basically the same. Except I guess its more like 'upgrade istiod to n+1' (while leaving ztunnel behind).

> I'm not sure under what circumstances a waypoint would need compatibility with a different Istiod, since the deployment is controlled by Istiod..

For canary, never. But we do advertise n-1 support for in-place users, so this would be nice to have I guess

> For Ztunnel/Istiod compat, do I just run connectivity tests, or do I need to look into WDS?

We will want to verify its not rejecting config I guess. A common false-negative would be we upgrade the control plane, its totally broken, but ztunnel has the last know state.

Maybe even a ztunnel restart thrown in would be a good measure

---

Basically all ideas above can be incremental improvements once we have the core structure
